src/scheduler/pbs_sched.c

```diff
@@ -791,46 +502,285 @@ are_we_primary()
 	return -1;		/* cannot be neither */
 }
 
+static void
+close_server(svr_t *sconn)
+{
+	if (sconn) {
+		char *svrhost = sconn->svrhost;
+
+		delete_link(&sconn->all_svrs_link);
+		tpp_em_del_fd(poll_context, sconn->sfd);
+		pbs_disconnect(sconn->pfd);
+		pbs_disconnect(sconn->sfd);
+		if (sconn->cmd.jid)
+			free(sconn->cmd.jid);
+		free(sconn);
+		sconn = NULL;
+		log_eventf(PBSEVENT_ADMIN | PBSEVENT_FORCE, PBS_EVENTCLASS_SCHED, LOG_INFO,
+			   msg_daemonname, "Disconnected from server %s", svrhost ? svrhost : pbs_default());
+	}
+}
+
+static void
+close_servers(void)
+{
+	svr_t *cur, *next;
+
+	cur = GET_NEXT(servers);
+	while (cur) {
+		next = GET_NEXT(cur->all_svrs_link);
+		close_server(cur);
+		cur = next;
+	}
+	tpp_em_destroy(poll_context);
+	CLEAR_HEAD(servers);
+}
+
+static int
+connect_server_helper(char *host, int is_primary)
+{
+	int rc;
+	int fd = -1;
+	struct batch_reply *reply = NULL;
+	char *conn_type = is_primary ? "primary" : "secondary";
```

Comment:
why? this is as per design, and moreover what will gain by using enum?

Reply:
The gain is readability and also we can avoid string comparison.

Reply:
As I have explained as part of other comment also, a) this is design change b) I don't mind to change it to enum but will not gain anything out of it as using explicit string is more readable then enum and c) they won't be much string comparison, its just 2 times for one sched

Reply:
I'd vote for the enum because I generally shy away from strings in C.  If we were in python, I'd be fine with it.

Reply:
conn type is removed from register batch request, as now first connection will be primary and second connection will secondary connection, so now need of string or enum and this comment is invalid

Reply:
I actually liked the fact that the batch request registered a connection as primary or secondary.  We've run into trouble before when the scheduler and server got out of sync.  Think of a case when the scheduler makes the primary connection, crashes, restarts, and then makes the primary connection again.  The server will mark the second connection as secondary, and then when the scheduler connects for the third time, it will be rejected.  If the scheduler registers two primary connections, you can discard the first and then everything is still fine.